src/util.h

```diff
@@ -525,6 +526,7 @@ class ArrayBufferViewContents {
   T stack_storage_[kStackStorageSize];
   T* data_ = nullptr;
   size_t length_ = 0;
+  bool was_detached_ = false;
```

Comment:
Why is it necessary to store this? It probably doesn't matter much, but most code that uses `ArrayBufferViewContents` doesn't need this and explicitly getting its value has the same cost.

If it's too verbose to add something like `value->ArrayBuffer() && value.As<ArrayBuffer>()->WasDetached()` in the caller itself, then I'd put that in a helper method along the lines of `bool IsDetachedArrayBuffer(v8::Local<v8::Value> value) { ... }` and call that then.

Reply:
From what I understood, and what @bnoordhuis recommended the reason to use ArrayBufferViewContents is to give an easier API (on top of other performance related reasons) to interact with ArrayBuffer contents. Adding `WasDetached()` on top of it, would make C++ developers job a lot easier. From where I'm sitting, the only downside of having a stored value for `WasDetached` is 1-byte, and I think that we need to check for detachedness in most of these places (IMHO, this is more like a hunch rather than a fact). Please correct me if I misinterpreted anything.